docs/data/ampli/migration.md

```diff
@@ -4,34 +4,305 @@ description: If you already have analytics implemented in your product you can e
 
-## Migrate events 
+#### Code Example
 
-For example, if you're currently tracking events like:
+```typescript
+import * as amplitude from '@amplitude/analytics-browser';
 
-```ts
-analytics.track('Song Uploaded',{
-  device: 'iOS'
+amplitude.init(AMPLITUDE_API_KEY);
+amplitude.track('Song Played', { title: 'The best song in the world.' });
+amplitude.track({
+  event_type: 'Song Played',
+  event_properties: {
+    title: 'Song 2'
+  }
 });
+amplitude.flush();
 ```
 
-After you import your tracking plan into Amplitude Data, this event becomes the following, with all the benefits of type-safety, linting, and data validation to make sure that your analytics are accurate and trustworthy.
+### Step 1 - Amplitude SDK & Ampli Together
 
-```ts
-ampli.songUploaded({
-  device: 'iOS'
+It is easy to use Ampli with your existing Amplitude SDK implementation. Learn more about how Ampli and the Amplitude SDK work together in the [Ampli overview](/data/sdks/ampli-overview) documentation.
+
+Start by passing your existing Amplitude SDK instance to Ampli. Ampli will use the given instance for all event tracking. This includes the API key, plugins, storage provider, log provider and other configuration set on the Amplitude instance.
+
+```typescript
+ampli.load({ client: { instance: amplitude }})
+```
+
+All existing `amplitude.track('Song Played')` (a.k.a `amplitude.logEvent()`) will continue to work. However, you now also have access to strongly typed methods and types for all events for the Source in your tracking plan e.g. `ampli.songPlayed()` and `ampli.track(new SongPlayed())`.
+
+#### Process
+
+!!!info "Use Branching for easier Migration"
+    We recommend creating a branch and adding small sets of events at a time. This splits the migration into manageable parts rather than requiring all events be implemented at once.
+
+    If you are planning to use `ampli status` to verify instrumentation, all events for the current source must be implemented to pass. Isolating the instrumentation into small parts allows you to use `ampli status` at each part of the migration versus waiting for all events to be completed.
+
+1. Add some events to the tracking plan in Data. Add those events to your Source.
+2. `ampli pull` to download the generated Events to your project
+3. Start replacing `amplitude.track('My Event', {prop: true})` with `ampli.myEvent({ prop: true})` or `ampli.track(new MyEvent({prop:true}))` for each event in the tracking plan.
+4. Repeat until all Amplitude SDK calls have been replaced by Ampli.
+
+#### Code Example
+
+```diff
+import * as amplitude, { BaseEvent } from '@amplitude/analytics-browser';
++ import { ampli, SongPlayed } from './path/to/ampli';
+
+// Original Implementation
+// Notice this will keep working as-is, so you can keep the
+// existing implementation while progressingly migrating to Ampli
+
+amplitude.init('API-KEY');
+amplitude.add(new MyPlugin());
```

Comment:
Not related to these changes, but can we link the [plugin doc](https://pr-760.d19s7xzcva2mw3.amplifyapp.com/data/sdk-plugins/#destination-type-plugin) in ampli docs? Maybe Ampli overview is the best fit.